xoinvader/ship.py

```diff
@@ -297,6 +303,16 @@ def __init__(self, pos, border, settings):
         self._pos = Point(
             x=pos.x - self._image.width // 2,
             y=pos.y - self._image.height)
+
+        self._collider = Collider("playership",
+                                  ["  #  ",
+                                   "#####",
+                                   " # # "],
+                                  self._pos)
+        def enemy_clash(_):
+            self.take_damage(10)
+
+        self._collider.add_handler("x_enemy", enemy_clash)
```

Comment:
Hmmm... I though there might be a situation where we want to process collisions of instances of same class differently.

But I reconsidered.  It seems like unnecessary complication now. So you're right.